content/docs/jsx-in-depth.md

```diff
@@ -247,6 +247,39 @@ function App2() {
 
 Spread attributes can be useful when you are building generic containers. However, they can also make your code messy by making it easy to pass a lot of irrelevant props to components that don't care about them. We recommend that you use this syntax sparingly.
 
+An alternative to spreading all of your props is to specifically list out all the properties that you would like to consume, followed by `...other`.
+
+```js
+const { children, ...other } = props;
+```
+
+This ensures that you pass down all the props *except* the ones you're consuming yourself.
+
+```js{6}
+const handleClick = () => console.log("handleClick");
+
+const PrimaryButton = props => {
+  const { children, ...other } = props;
```

Comment:
> Could we pick a different example? Since in this case `type` would be passed the same regardless of whether you plucked it out of props or passed it via `other`. - @bvaughn

To your [original comment](https://github.com/reactjs/reactjs.org/pull/121#discussion_r143798264), I updated the example removing `type` from this line. I wasn't trying to focus on `type` but I was trying to show how `PrimaryButton` could now receive unknown props like `onClick`

```jsx
<PrimaryButton type="button" onClick={handleClick}>
```

If this still isn't preferrable, I can just use the original example as-is from the original docs ([Transferring with `...` in JSX](https://github.com/facebook/react/blob/v15.4.0-rc.3/docs/docs/06-transferring-props.md#transferring-with--in-jsx))

Reply:
I think this is better! But it's still a little awkward since you _could_ also pass `children` via the spread operator as well- and `<button type="button">` isn't a valid attribute. These might be confusing to readers. (They might get distracted thinking, "why is that..." instead of focusing on the spread operator.)

How do you feel about this?
```jsx
const handleClick = () => console.log("handleClick");

const PrimaryButton = props => {
  const { type, ...other } = props;
  const className = type === "primary" ? "PrimaryButton" : "SecondaryButton";
  return <button className={className} {...other} />;
};

const App = () => {
  return (
    <div>
      <PrimaryButton type="primary" onClick={handleClick}>
        Hello World!
      </PrimaryButton>
    </div>
  );
};
```

Reply:
> `<button type="button">` is a valid attribute

You are correct, of course. Sometimes I forget how wonky HTML attributes are 😆 

```js
const { ...other } = props;
return <button {...other} />;
```

I think this maybe leaves too much to the imagination. "Why would I do this? What's the point?" I think we should show at least one prop being plucked out.

Reply:
😆Alright, with everything said so far, 

I can pluck `className` instead of `type` to avoid confusion 
(hopefully, I'm using the "pluck" terminology correctly)

The `PrimaryButton` component takes a `className` prop and any number of unknown props through the `...other` object, just as long as they are valid HTML attributes for `<button>`. 

```jsx
const handleClick = () => console.log("handleClick");

const PrimaryButton = props => {
  const { className, ...other } = props;
  return <button className={className} {...other} />;
};

const App = () => {
  return (
    <div>
      <PrimaryButton className="primaryButton" onClick={handleClick}>
        Hello World!
      </PrimaryButton>
    </div>
  );
};
```
Now, the `App` component can return the `PrimaryButton` so that it can receive it's known props (`className`) *and* any unknown props through `...other`, which contains the following when used in `App`: 

```js
// Contained in { ...other }
{ 
  onClick: handleClick(), 
  children: 'Hello World!' 
}
```

Reply:
I would have the same concern as I mentioned before:
```js
// This...
const PrimaryButton = props => {
  const { className, ...other } = props;
  return <button className={className} {...other} />;
};
// Is the same thing as this...
const PrimaryButton = props => {
  return <button {...props} />;
};
```